sphinx/util/i18n.py

```diff
@@ -143,10 +146,7 @@ def find_catalog_source_files(locale_dirs, locale, domains=None, gettext_compact
                 if excluded(path.join(relpath(dirpath, base_dir), filename)):
                     continue
                 base = path.splitext(filename)[0]
-                domain = relpath(path.join(dirpath, base), base_dir)
-                if gettext_compact and path.sep in domain:
-                    domain = path.split(domain)[0]
```

Comment:
When `gettext_compact` is True, the `domain` sometime points non-existence catalog. The compaction is only needed on looking up catalogs for document.
So I'm dropping this.